architecture/kubernetes_infrastructure.adoc

```diff
@@ -9,30 +9,43 @@
 
-The host(s) containing the master components (API Server, Controller Manger Server, etcd, etc)
+A Kubernetes cluster involves a master and a set of nodes. 
 
-=== API Server
+== Master
 
-Serves the main API
+The Kubernetes master is the host or hosts that contain the master components (API Server, Controller Manger Server, etcd, etc)
 
-=== Controller Manager Server
+=== API Server
 
-Watches etcd for changes to ReplicationController objects and then uses the API to enforce the desired state.
+The Kubernetes API server serves the main API, and validates and configures the data for pods, services, and replication controllers. The Kuberbetes API server also assigns pods to nodes and synchronizes pod information with service configuration. 
 
 === etcd
 
 All persistent master state is stored in etcd. Other components watch etcd for changes and eventually bring themselves into the desired state described in etcd.
 
+=== Controller Manager Server
+
+The controller manager server watches etcd for changes to replication controller objects and then uses the API to enforce the desired state.
+
 == Node
 
-Each node has the services necessary to run Docker containers and be managed from the master systems
+A Kubernetes node is managed from the master, and has the services necessary in order to run Docker containers. Each node runs Docker, which downloads and runs containers.
+
+A node is not created by Kubernetes, it is instead created by your cloud provider, or from your physical or virtual machines. Once Kubernetes is involved, it will validate the node with health checks. A node will be ignored until these health checks are passed. Note that Kubernetes will keep checking nodes until they are valid. 
 
 === Kubelet
 
-Each node has a kubelet that works in terms of a container manifest. A container manifest is a YAML file that describes a pod. The kubelet takes a set of manifests and ensures that the containers described in those manifests are started and continue running.
+Each node has a kubelet that updates the node as specified by a container manifest; a YAML file that describes a pod. The kubelet uses a set of manifests to ensure that the containers described start and continue to run. A sample manifest can be found https://cloud.google.com/compute/docs/containers/container_vms#container_manifest[here].
+
+There are a number of ways that a container manifest can be provided to a kubelet:
+
+* A file path on the command line. The file is checked every twenty seconds.
+* A HTTP endpoint passed on the command line. The endpoint is checked every twenty seconds.
+* The kubelet will watch an etcd server, such as `[filename]#/registry/hosts/$(hostname -f)#`. Any changes are noticed and acted upon.
+* The kublet will listen for HTTP and respond to a simple API tp submit a new manifest.
```

Comment:
s/kublet/kubelet
s/tp/to

The kubelet listens for HTTP and responds to a simple API to submit a new manifest.